e2e/__tests__/transform.test.ts

```diff
@@ -185,3 +185,14 @@ describe('transformer-config', () => {
     expect(stdout).toMatchSnapshot();
   });
 });
+
+describe('transform-environment', () => {
+  const dir = path.resolve(__dirname, '..', 'transform/transform-environment');
+
+  it('should transform the environment', () => {
```

Comment:
I'm not sure how this asserts that the environment was transformed? Maybe we could add some TS types to the custom env you wanna transpile or something?

Reply:
For now I've modified it such that the environment will have to be transformed by `babel-jest` as it uses `import`

Reply:
I was thinking maybe using preset: `ts-jest`, so as to have the exact scenario mentioned in the issue.

I'm not sure how dependencies in e2e test work, I've added it in my `package.json`
of the folder corresponding to the test. Do I do a `yarn install` within the e2e folder?
Think seemed odd to do it inside a test folder. And how does this work in CI?

Reply:
We have a few tests installing deps. Just look for `yarn.lock` files in the e2e

Reply:
Putting TS types on it is probably better since that's less likely than ESM to be natively supported by the runtime at some point.
A few e2e tests do yarn installs `beforeEach` or so, but I believe many also rely on things like `@babel/preset-typescript` being present in the root, so that'd be totally fine here I think.

Reply:
I'll add TS types and try use `@babel/preset-typescript` for the preprocess and let you know

Reply:
The test has been updated as per the suggestions. The environment is in typescript, will also be passing a variable to the test by setting in `this.global`